Doxyfile

```diff
@@ -0,0 +1,115 @@
+# Doxyfile 1.9.4
+
+# This file describes the settings to be used by the documentation system
+# doxygen (www.doxygen.org) for a project.
+#
+# All text after a double hash (##) is considered a comment and is placed in
+# front of the TAG it is preceding.
+#
+# All text after a single hash (#) is considered a comment and will be ignored.
+# The format is:
+# TAG = value [value, ...]
+# For lists, items can also be appended using:
+# TAG += value [value, ...]
+# Values that contain spaces should be placed between quotes (\" \").
+#
+# Note:
+#
+# Use doxygen to compare the used configuration file with the template
+# configuration file:
+# doxygen -x [configFile]
+# Use doxygen to compare the used configuration file with the template
+# configuration file without replacing the environment variables:
+# doxygen -x_noenv [configFile]
+
+DOXYFILE_ENCODING      = UTF-8
+PROJECT_NAME           = "Endless Sky"
+PROJECT_NUMBER         =
+PROJECT_BRIEF          = "This is an auto generated documentation for the source code of endless-sky."
+PROJECT_LOGO           = icons/icon_48x48.png
+OUTPUT_DIRECTORY       = ./docs/api
+
+CREATE_SUBDIRS         = YES
+
+INPUT                  = ./source
+RECURSIVE              = YES
+EXCLUDE                =
+EXCLUDE_SYMLINKS       = YES
+
+EXTRACT_ALL            = YES
+EXTRACT_PRIVATE        = YES
+EXTRACT_PRIV_VIRTUAL   = YES
+EXTRACT_STATIC         = YES
+EXTRACT_LOCAL_CLASSES  = YES
+EXTRACT_ANON_NSPACES   = YES
+
+WARN_NO_PARAMDOC       = YES
+WARN_AS_ERROR          = NO
+WARN_LOGFILE           =
+
+CLANG_ASSISTED_PARSING = NO
```

Comment:
> If the CLANG_ASSISTED_PARSING tag is set to YES then Doxygen will use the [clang parser](http://clang.llvm.org/) for more accurate parsing at the cost of reduced performance. This can be particularly helpful with template rich C++ code for which Doxygen's built-in parser lacks the necessary type information.

This sounds like a useful option.

Reply:
Is the current parsing insufficiently accurate so that we need this? Or is this more of a nice-to-have?

I'm not opposed to enabling this option, but I feel that setting it up will be quite some additional effort. (And I would like to keep that for a follow-up PR.)

Reply:
If it's a lot of effort, I suppose it could be in a follow-up. It did seem to work when I set it to "YES".

Reply:
It's now on, and I added an option for c++20, because it needed to switch over to c++20.